mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -70,6 +88,17 @@ object MvRxViewModelProvider {
                     "single non-optional parameter that takes initial state of ${stateClass.simpleName}."
             }
         }
+
+        viewModelContext.savedStateRegistry.registerSavedStateProvider(key) {
+            withState(viewModel) { state ->
+                Bundle().apply {
+                    putBundle(KEY_MVRX_SAVED_INSTANCE_STATE, state.persistState())
+                    putParcelable(KEY_MVRX_SAVED_ARGS, viewModelContext.args as Parcelable?)
```

Comment:
there isn't a specific interface that the arguments conform to, is there a way to store in the bundle this without switching on the type?

Reply:
I guess before we were treating it as a serializable.

I _think_ it is safe to treat it as either a serializable or a parcelable, so you could use a `when` statement to check for those two cases (primitives, lists, and strings would fall under the Serializable case), and I suppose it would be best to throw in the else case so we are aware of issues